.github/workflows/build_push_image.yaml

```diff
@@ -0,0 +1,71 @@
+name: Build and push container image
+
+on:
+  # Manual trigger for the workflow.
+  workflow_dispatch:
+    inputs:
+      release_tags:
+        required: false
+        default: 'next'
+
+  # Every day at 01:32 AM UTC.
+  schedule:
+    - cron: "32 01 * * *"
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      # This is kept intentionally separate from "Set environment".
+      - name: Check scheduled run
+        if: github.event_name == 'schedule'
+        run: |
+          last_commit=$(git log -n1 --since=yesterday --oneline)
+          if [ -z "${last_commit}" ]; then
+              echo "No new commits found, cancelling ..."
+              exit 1
+          else
+              echo "Proceeding with the current tip:"
+              echo "  ${last_commit}"
+          fi
+
+      - name: Set environment
+        run: |
+          if [ "${{ github.event_name }}" = "workflow_dispatch" ]; then
+              echo "RELEASE_TAGS=${{ github.event.inputs.release_tags }}" >> $GITHUB_ENV
+          elif [ "${{ github.event_name }}" = "schedule" ]; then
+              echo "RELEASE_TAGS=next" >> $GITHUB_ENV
+          elif [ "${{ github.event_name }}" = "release" ]; then
+              :
+          fi
+
+      - name: Log in to the registry
+        uses: redhat-actions/podman-login@v1
+        with:
+          registry: quay.io/retis
+          username: ${{ secrets.REGISTRY_USERNAME }}
+          password: ${{ secrets.REGISTRY_TOKEN }}
```

Comment:
Sorry I only see this now, I think this can be done at the `redhat-actions/push-to-registry@v2` stage by adding `username` and `password` under `with:`.

Reply:
yes, it could, but I split to make it bail it early (without building) in case of issues (network failures and so forth).
Do you prefer to squash this in `push-to-registry`

Reply:
> yes, it could, but I split it to make it bail it early (without building) in case of issues (network failures and so forth).

Makes sense.

> Do you prefer to squash this in `push-to-registry`?

The advantage is building stages are not logged in; not sure if that is really better. So up to you :)

Reply:
had no strong opinion on it, eventually, I went with your suggestion.
I triggered a manual build on my repo to check the upload goes fine, just in case.
I'll put the link here once the job's done.

Reply:
Job successfully ran to completion:

https://github.com/vlrpl/retis/actions/runs/11387554527/job/31682439658

Reply:
Thanks!